daemon/volumes.go

```diff
@@ -52,6 +52,9 @@ func (m *mountPoint) Setup() (string, error) {
 			if !os.IsNotExist(err) {
 				return "", err
 			}
+			if _, err := os.Stat(m.Source); err != nil && os.IsNotExist(err) {
```

Comment:
This needs to check for an error that is not a `os.IsNotExist` and handle that appropriately:

```
if fi, err := os.Stat(m.Source); err != nil {
 if !os.IsNotExist(err) {
     //.. handle error 
 }
 logrus.Warnf("Auto-creating non-existant volume host path %s, this is deprecated and will be removed soon", m.Source)
}
```